compiler/noirc_driver/src/lib.rs

```diff
@@ -141,6 +141,10 @@ pub struct CompileOptions {
     #[arg(long)]
     pub skip_brillig_constraints_check: bool,
 
+    /// Flag to turn off preprocessing functions during SSA passes.
+    #[arg(long)]
+    pub skip_preprocess_fns: bool,
```

Comment:
Why do we need this?

Reply:
1) It made it very convenient to check the effect of doing or not doing any preprocessing on the circuit size
2) I wasn't sure if it wouldn't make things slower and had to be turned off for some reason

It looks like we didn't have too much adverse affect, so we can remove it, but it was nice for 1) while we experiment with it.

Reply:
Yeah, I don't see a reason to turn off preprocessing at this point so I'd say we can remove it.